parser/src/parser.rs

```diff
@@ -145,7 +146,14 @@ impl<'a> Parser<'a> {
         match pivot {
             If => self.parse_if(Parser::statement).map(Expr::If),
             Match => self.parse_match(Parser::statement).map(Expr::Match),
-
+            Identifier
+                if self
+                    .cursor
+                    .lookahead(of_type(Identifier).and_then(of_type(RoundedLeftBracket)))
+                    .is_some() =>
+            {
+                self.constructor()
+            }
```

Comment:
isn't it a bit weird that this exact code section also figures in the `Parser::next_value` ? maybe you would like to create your own `newt_x`(let's call that a 'pivot') method,

Reply:
Yep totally